lib/src/services/request_cache.dart

```diff
@@ -1,44 +1,33 @@
 import 'dart:collection';
+import 'package:dcache/dcache.dart';
+import 'package:http/http.dart';
 
-import 'package:http/http.dart' as http;
-
-import '../models/node.dart';
+import 'typedefs.dart' as defs;
 
 /// Cache store which uses a [HashMap] internally to serve requests.
 class RequestCache {
-  final _cachedResponses = HashMap<int, _Cache>();
+  Cache _cachedResponses;
+  final Duration timeToUse;
+  final int size;
+  final defs.Send<Map<String, dynamic>> send;
+
+  RequestCache(this.size, this.timeToUse, this.send) {
+    _cachedResponses = LruCache<dynamic, Response>(storage: InMemoryStorage(size));
+  }
 
+  // TODO(harisarang): rename this function to getResponse
   /// Caches the response of the [request], identified by [key]. The cached
   /// response is valid till [cacheTTL].
-  Future<Map<String, dynamic>> cache(
+  Future<Map<String, dynamic>> getResponse(
     int key,
-    Future<Map<String, dynamic>> Function(Future<http.Response> Function(Node))
-        send,
-    Future<http.Response> Function(Node) request,
-    Duration cacheTTL,
+    defs.Request request,
   ) async {
     if (_cachedResponses.containsKey(key)) {
-      if (_isCacheValid(_cachedResponses[key], cacheTTL)) {
-        // Cache entry is still valid, return it
-        return Future.value(_cachedResponses[key].data);
-      } else {
-        // Cache entry has expired, so delete it explicitly
-        _cachedResponses.remove(key);
-      }
+      return send(_cachedResponses.get(key));
```

Comment:
I think this line should've given you an error. The `send` function only takes in a `request`.